UPGRADE-4.0.md

```diff
@@ -0,0 +1,91 @@
+# Upgrade from v3.x to v4.x
+
+If you have a question regarding this upgrade process, please check code in `examples` directory:
+
+- [Complete](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/tree/master/examples/complete)
+
+If you find a bug, please open an issue in this repository including configuration to reproduce.
+
+:warning: Note - due to the nature of the default use of `name_prefix`, nearly al paths to upgrade will require recreating the resources provisioned by the module. For most use cases this should be ok as autoscaling groups are intended to be ephemeral, but nonetheless, please be aware.
```

Comment:
`al` => `all`

Reply:
darn, looks like my spell checker turned off - fixed in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/139/commits/0440363ae5a8ba3c4126a31d63f5794bf3c53947